### PR TITLE
MYFACES-4459 Update poms javax -> jakarta

### PR DESCRIPTION
**Pull request by volosied**
https://issues.apache.org/jira/projects/MYFACES/issues/MYFACES-4459

Still in progress, but I wanted an initial PR up.

### CODE REVIEW

**Review by melloware — APPROVED** [review]

**Comment by volosied** [comment]
I looked at the new MANIFEST, and I don't see the versions for some of the imported packages. (such as jakarta.el and jakarta.servlet) I compared against 3.0.x which does export the versions. Not sure what's going on. I'll need to compare POMs.

**Comment by volosied** [comment]
Believe the versions are due to the jars themselves. I've reproduced this issue with this pom: 

```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">

    <groupId>test.manifest</groupId>
    <artifactId>manifiest</artifactId>
    <version>1.0.0-SNAPSHOT</version>
    <name>MissingImportVersionsInManifest</name>

    <modelVersion>4.0.0</modelVersion>

    <build>

        <plugins>
            <!-- create manifest.mf for OSGi -->
            <plugin>
                <groupId>org.apache.felix</groupId>
                <artifactId>maven-bundle-plugin</artifactId>
                <executions>
                    <execution>
                        <id>bundle-manifest</id>
                        <phase>process-classes</phase>
                        <goals>
                            <goal>manifest</goal>
                        </goals>
                        
                        <configuration>
                            <instructions>
                                <Import-Package>  
                                jakarta.el;version="[5.0.0, 6.0.0)"
                                </Import-Package>
                            </instructions>
                        </configuration>
                    </execution>
                </executions>
            </plugin>

        </plugins>

    </build>


    <dependencies>
        <!-- WORKING : Import-Package: jakarta.el;version="[5.0.0,6.0.0)" -->
        <!-- <dependency>
            <groupId>jakarta.el</groupId>
            <artifactId>jakarta.el-api</artifactId>
            <version>5.0.1</version>
            <scope>provided</scope>
        </dependency> -->
        <!-- ^^^ Jakarta API Manifest -> Export-Package: jakarta.el;version="5.0.1" -->

        <!-- FAILING: Import-Package: jakarta.el -->
        <dependency>
            <groupId>org.apache.tomcat</groupId>
            <artifactId>tomcat-el-api</artifactId>
            <version>10.1.0</version>
             <scope>provided</scope>
        </dependency>
        <!-- ^^^^ Tomcat API Manifest -> Export-Package: jakarta.el;version="5.0" -->

    </dependencies>

</project>
```

**Comment by volosied** [comment]
1)     Created https://issues.apache.org/jira/browse/FELIX-656 and downgraded the maven-bundle-plugin to 4.0.0  to get the versions outputted again.
2)  Updated the EE dependencies to the latest

File: impl/pom.xml [hunk 1]
```diff
@@ -629,17 +629,16 @@
...
+                                    jakarta.annotation,
                                     javax.crypto,
                                     javax.crypto.spec,
-                                    javax.ejb;resolution:=optional,
-                                    javax.el;version="[1.0.0, 3.1.0)",
+                                    jakarta.ejb;resolution:=optional,
+                                    jakarta.el;version="[5.0.0, 6.0.0)",
                                     javax.naming,
-                                    javax.persistence;version="[1.0.0, 2.1)";resolution:=optional,
-                                    javax.portlet;version="[1.0.0, 2.1)";resolution:=optional,
```
Comment (volosied) [inline impl/pom.xml:639]: @pnicolucci @tandraschko  Any objections to removing javax.portlet as there's no EE9 and EE10 equivalent?  

However, the faces 4.0 spec still mentions portlet.    

1) Remove portlet import
2) change to jakarta.portlet 
3) leave as javax.portlet (no change) 


With option two, there's one portlet search hit here: https://github.com/apache/myfaces/blob/e9fe59f96410f31a7f5c0fbd6838c1a22683a691/impl/src/main/java/org/apache/myfaces/util/ExternalContextUtils.java#L41

File: bundle/pom.xml [hunk 2]
```diff
@@ -67,21 +67,20 @@
...
-                            javax.servlet.jsp.jstl.core;version="[1.1.2, 2.0.0)",
-                            javax.servlet.jsp.tagext;version="[2.1.0, 3.1)",
-                            javax.servlet.annotation;version="[3, 5)";resolution:=optional,
-                            javax.validation.*;version="[1.0.0, 2.0.0)";resolution:=optional,
+                            jakarta.persistence;version="[3.1.0, 4.0.0)";resolution:=optional,
+                            jakarta.servlet;version="[6.0.0, 7.0.0)",
+                            jakarta.servlet.http;version="[6.0.0, 7.0.0)",
+                            jakarta.servlet.jsp;version="[6.0.0, 7.0.0)",
+                            jakarta.servlet.jsp.jstl.core;version="[3.0.0, 4.0.0)",
+                            jakarta.servlet.jsp.tagext;version="[3.1.0, 4.0.0)",
```
Comment (volosied) [inline bundle/pom.xml:81]: Should I remove  `jakarta.servlet.jsp` , `jakarta.servlet.jsp.jstl.core`, and `jakarta.servlet.jsp.tagext` ? 

There are no hits for these packages in the 4.0 code base.  Only hits I see are in `extensions/quarkus/runtime/pom.xml`

Comment (melloware) [reply bundle/pom.xml:81]: Yep see: https://github.com/apache/myfaces/pull/316

Comment (volosied) [reply bundle/pom.xml:81]: I don't see them in Mojarra either.  Will remove then :) 


Mojarra (org.glassfish javax.faces 4.0.0) Imports:

 jakarta.el;version="[5.0,6)",

 jakarta.servlet.annotation;resolution:=optional;version="[6.0,7)",
 jakarta.servlet;version="[6.0,7)",
 jakarta.servlet.http;version="[6.0,7)",

 jakarta.validation;resolution:=optional;version="[3.0,4)",
 jakarta.validation.groups;resolution:=optional;version="[3.0,4)",

  jakarta.annotation;version="[2.1,3)",

  jakarta.ejb;resolution:=optional;version="[4.0,5)",

  jakarta.persistence;resolution:=optional;version="[3.1,4)",

File: impl/pom.xml (hunk 1 above)
Comment (volosied) [reply impl/pom.xml:639]: Mojarra doesn't have any portlet imports either. I think option 1 is best. I created https://issues.apache.org/jira/browse/MYFACES-4472 as a follow up

Comment (pnicolucci) [reply impl/pom.xml:639]: My vote would be to keep it as it until MYFACES-4472 is investigated and completed.

Comment (volosied) [reply impl/pom.xml:639]: I've added it back in.

**Review by pnicolucci — APPROVED** [review]

**Pull request merged**